router-tests/telemetry_test.go

```diff
@@ -2267,4 +2267,103 @@ Downstream errors:
 			require.Equal(t, "exception", events[0].Name)
 		})
 	})
+
+	t.Run("Datadog Propagation", func(t *testing.T) {
```

Comment:
Please also add tests where the client sends DD trace headers.

Reply:
added